common/src/KokkosKernels_SimpleUtils.hpp

```diff
@@ -151,10 +151,10 @@ inline void kk_exclusive_parallel_prefix_sum(
 template <typename forward_array_type, typename MyExecSpace>
 void kk_inclusive_parallel_prefix_sum(
     typename forward_array_type::value_type num_elements,
-    forward_array_type arr) {
-  typedef Kokkos::RangePolicy<MyExecSpace> my_exec_space;
+    forward_array_type arr, MyExecSpace my_exec_space = MyExecSpace()) {
```

Comment:
The normal pattern when allowing an execution space instance in a kernel is to do the following:
```
// Add a new overload with the execution space instance as first parameter
// this follows the patter of Kokkos Core and makes our library uniform, i.e.
// user friendly
template <typename forward_array_type, typename MyExecSpace>
void kk_inclusive_parallel_prefix_sum(MyExecSpace space, ...) {}

// Replace the old implementation with a simple wrapper that
// creates an instance of the execution space and passes it to
// the new overload, this is for backward compatibility
template <typename forward_array_type, typename MyExecSpace>
void kk_inclusive_parallel_prefix_sum(...) {
  MyExecSpace space{};
  kk_inclusive_parallel_prefix_sum(pace, ...);
}
```

In the case of `kk_inclusive_parallel_prefix_sum` since the function is in the impl namespace we can be a bit more lose and we can remove backward compatibility since it should only be used in the library but you might want to check for any usage in Trilinos...